File: Java/jobspls/algorithms/BinaryGap.java
package jobspls.algorithms;

public class BinaryGap {
    public static int binary_gap(int i) {
        var binary = Integer.toBinaryString(i);
        var gap = 0;
        var current = 0;

        for (var b : binary.toCharArray()) {
            if (b == '1') {
                if (current > gap) {
                    gap = current;
                }
                current = 0;
            } else {
                ++current;
            }
        }

        return gap;
    }
}

File: Go/jobspls/algorithms/binary_gap.go
package algorithms

import "fmt"

func BinaryGap(i int) int {
	binary := fmt.Sprintf("%b", i)
	gap := 0
	current := 0

	for _, n := range binary {
		if n == '1' {
			if current > gap {
				gap = current
			}
			current = 0
		} else {
			current++
		}
	}

	return gap
}
